packages/gatsby-recipes/src/providers/gatsby/plugin.js

```diff
@@ -299,56 +320,113 @@ class BabelPluginAddPluginsToGatsbyConfig {
             )
 
             if (shouldAdd) {
-              const plugins = pluginNodes.value.elements.map(getPlugin)
-              const matches = plugins.filter(plugin => {
-                if (!key) {
-                  return plugin.name === pluginOrThemeName
-                }
-
-                return plugin.key === key
-              })
+              if (t.isCallExpression(pluginNodes.value)) {
+                const plugins = pluginNodes.value.callee.object.elements.map(
+                  getPlugin
+                )
+                const matches = plugins.filter(plugin => {
```

Comment:
`matches` is only used as a conditional expression. Suggest to use `.some` or `.every` here instead. (Yeah the other branch has the same issue..)